### PR TITLE
Refactor logging (backport #543)

### PR DESCRIPTION
**Pull request by christophfroehlich**
#543 with additional local variables/methods for logging, which are available from jazzy on only.

### CODE REVIEW

**Comment by christophfroehlich** [comment]
@mergifyio backport iron

File: example_14/hardware/rrbot_actuator_without_feedback.cpp [hunk 1]
```diff
@@ -220,18 +217,19 @@ hardware_interface::return_type ros2_control_demo_example_14::RRBotActuatorWitho
...
 
-    // Simulate sending commands to the hardware
     std::ostringstream data;
     data << hw_joint_command_;
-    RCLCPP_INFO(
-      rclcpp::get_logger("RRBotActuatorWithoutFeedback"), "Sending data command: %s",
-      data.str().c_str());
-    send(sock_, data.str().c_str(), strlen(data.str().c_str()), 0);
+    ss << "Sending data command: " << data.str() << std::endl;
+    RCLCPP_INFO(get_logger(), ss.str().c_str());
```
Comment (saikishor) [inline example_14/hardware/rrbot_actuator_without_feedback.cpp:229]: @christophfroehlich is throttle missing here?

File: example_14/hardware/rrbot_sensor_for_position_feedback.cpp [hunk 2]
```diff
@@ -256,28 +249,27 @@ hardware_interface::CallbackReturn RRBotSensorPositionFeedback::on_deactivate(
...
-  RCLCPP_INFO(
-    rclcpp::get_logger("RRBotSensorPositionFeedback"), "Got state %.5f for joint '%s'!",
-    hw_joint_state_, info_.joints[0].name.c_str());
 
-  RCLCPP_INFO(rclcpp::get_logger("RRBotSensorPositionFeedback"), "Joints successfully read!");
+  ss << std::fixed << std::setprecision(2);
+  ss << "Got measured velocity " << measured_velocity << std::endl;
+  ss << "Got state " << hw_joint_state_ << " for joint '" << info_.joints[0].name << "'"
+     << std::endl;
+  RCLCPP_INFO(get_logger(), ss.str().c_str());
```
Comment (saikishor) [inline example_14/hardware/rrbot_sensor_for_position_feedback.cpp:272]: @christophfroehlich here as well?

File: example_6/hardware/rrbot_actuator.cpp [hunk 3]
```diff
@@ -155,15 +159,16 @@ hardware_interface::return_type RRBotModularJoint::read(
...
   hw_joint_state_ = hw_joint_state_ + (hw_joint_command_ - hw_joint_state_) / hw_slowdown_;
-  RCLCPP_INFO(
-    rclcpp::get_logger("RRBotModularJoint"), "Got state %.5f for joint '%s'!", hw_joint_state_,
-    info_.joints[0].name.c_str());
 
-  RCLCPP_INFO(rclcpp::get_logger("RRBotModularJoint"), "Joints successfully read!");
+  ss << std::fixed << std::setprecision(2) << std::endl
+     << "\t" << hw_joint_state_ << " for joint '" << info_.joints[0].name << "'";
+
+  RCLCPP_INFO(get_logger(), "%s", ss.str().c_str());
```
Comment (saikishor) [inline example_6/hardware/rrbot_actuator.cpp:171]: missing throttle here

File: example_6/hardware/rrbot_actuator.cpp [hunk 4]
```diff
@@ -173,14 +178,14 @@ hardware_interface::return_type ros2_control_demo_example_6::RRBotModularJoint::
...
 
   // Simulate sending commands to the hardware
-  RCLCPP_INFO(
-    rclcpp::get_logger("RRBotModularJoint"), "Got command %.5f for joint '%s'!", hw_joint_command_,
-    info_.joints[0].name.c_str());
+  ss << std::fixed << std::setprecision(2) << std::endl
+     << "\t" << hw_joint_command_ << " for joint '" << info_.joints[0].name << "'";
 
-  RCLCPP_INFO(rclcpp::get_logger("RRBotModularJoint"), "Joints successfully written!");
+  RCLCPP_INFO(get_logger(), "%s", ss.str().c_str());
```
Comment (saikishor) [inline example_6/hardware/rrbot_actuator.cpp:188]: missing throttle here

**Review by saikishor — CHANGES_REQUESTED** [review]

**Comment by christophfroehlich** [comment]
@saikishor we have the problems with throttling in the plugins, remember https://github.com/ros-controls/ros2_control_demos/pull/603 ;)

**Review by saikishor — APPROVED** [review]
You are right! Looks good to me

**Pull request merged**